lib/reverse_words.rb

```diff
@@ -1,4 +1,52 @@
+require "pry"
+
 # A method to reverse each word in a sentence, in place.
+
+def string_reverse(my_string)
+   if my_string == ""
+     return my_string
+   elsif my_string.nil?
+     return nil
+   else
+
+     length = my_string.length
+     count = length / 2
+     count.times do |i|
+       temp = my_string[i]
+       my_string[i] = my_string[length - i - 1]
+       my_string[length - i - 1] = temp
+     end
+   end
+   return my_string
+end
+
+
 def reverse_words(my_words)
-  raise NotImplementedError
+  if my_words.nil? || my_words == ""
+    return my_words
+  end
+
+  words = my_words.chars
+  length = words.length
+
+  words_split = my_words.split(" ")
```

Comment:
The problem statement asks you to not use any Ruby provided methods apart from .length. With that in mind, can you think of a solution without using .split?

Also, due to using .split, your space complexity will also be O(n). Since words_split and words_reversed will take up additional space.